generator.go

```diff
@@ -553,6 +553,23 @@ func processType(s *metaSchema, pName, pDesc, path, parentPath string) (typeRef
 				sf.TypePrefix = ""
 				sf.TypeRef = gotType
 				sf.PtrForOmit = true
+			} else if !hasAddlProps {
+				if patternProps := propSchema.PatternProperties; patternProps != nil {
+					var patternPropTypes []string
+					for _, prop := range patternProps {
+						patternPropTypes = append(patternPropTypes, fmt.Sprintf("%v", prop.Type))
+					}
+
+					for index, patternPropType := range patternPropTypes {
+						if index != 0 && patternPropType != patternPropTypes[index-1] {
+							log.Fatalln("can't properly parse this object, it's values should only have 1 type, but found 2:", patternPropType, "and", patternPropTypes[index-1])
+						}
+					}
+
+					if len(patternPropTypes) > 0 {
+						sf.TypePrefix = fmt.Sprintf("map[string]%s", patternPropTypes[0])
```

Comment:
can we just apply patternPropTypes like that ? what if everything is nil ?

Reply:
I guess if everything is nil the schema would not be valid since each property should have a `type` declared?

Reply:
But I need to check the length because else it will throw a warning that patternProptypes could be empty and we have a out of bounds memory error

Reply:
I don't think `"type"` is a required keyword, is it ?

Reply:
https://json-schema.org/understanding-json-schema/basics.html#the-type-keyword